### PR TITLE
core, eth, trie: assume code is always prefixed in db

### PR DESCRIPTION
**Pull request by holiman**
This PR removes the notion of having code without prefix in the database. 
We switched to using special code prefix quite a while ago, and this can be costly: in some cases, we do a double-check for with versus without prefix. 

TBH I'm not fully certain if we can change this now, or if we need to sit on it a while longer.

### CODE REVIEW

File: eth/protocols/eth/handlers.go [hunk 1]
```diff
@@ -180,10 +180,6 @@ func answerGetNodeDataQuery(backend Backend, query GetNodeDataPacket, peer *Peer
 			continue
 		}
 		entry, err := backend.Chain().TrieNode(hash)
-		if len(entry) == 0 || err != nil {
-			// Read the contract code with prefix only to save unnecessary lookups.
-			entry, err = backend.Chain().ContractCodeWithPrefix(hash)
-		}
```
Comment (holiman) [inline eth/protocols/eth/handlers.go:186]: This needs to be reverted

Comment (holiman) [reply eth/protocols/eth/handlers.go:186]: ( I thought we had a separate getCode message)

**Comment by karalabe** [comment]
Given that some people had old ancients with legacy format receipts in them, I'm fairly certain we have many nodes that have old format codes.

**Comment by holiman** [comment]
True dat

**Pull request closed without merging**